### ISSUE TITLE
Researcher needs to validate experiment design when saving an experiment to ensure valid data for the client application.

### ISSUE THREAD

**Issue by mswartzCL** [CarnegieLearningWeb/UpGrade#402]
**User Story**
 - **Researcher needs to validate decision points, IDs, and conditions when saving an experiment to ensure valid data for the client application.**

**Acceptance Criteria**
 - Scenario: **Context Metadata does not exist for AppContext**
     - Given **AppContext is selected for experiment and context metadata does not exist for that AppContext**, when **experiment is saved**, then **no validation will occur**.

 - Scenario: **Context Metadata validation is disabled globally**
     - Given **Context Metadata validation is disabled**, when **experiment is saved**, then **no validation will occur**.

 - Scenario: **Context Metadata exists for AppContext**
     - Given **AppContext is selected for experiment and context metadata exists for that AppContext**, when **experiment is saved**, then **decision points and conditions are validated against context metadata, and results will be displayed on the experiment detail screen**.

**Design**
[ - Link to design](https://www.figma.com/file/uYxnCT2dXTftSZKFQ00Q0y/UpGrade-UI-Prototype-(latest))
 
 **Context Metadata Model**
[Context Metadata Gist]( https://gist.github.com/danoswaltCL/2db2b62e09550a39f7cb02d725cf0290)

**Comment by mswartzCL** [comment]
@zackcl Can you please link your relevant design work? Thanks!

**Comment by zackcl** [comment]
Here's a link to the latest Figma design: https://www.figma.com/file/uYxnCT2dXTftSZKFQ00Q0y/UpGrade-UI-Prototype-(latest)

**Comment by zackcl** [comment]
I don't think we need to validate the context metadata and provide feedback (e.g. showing error icons) when creating an experiment though the stepper UI because the fields that need validation (e.g. Site, Target, Condition) can be completed using the combo box, and the available options will be pulled from the metadata so it guarantees the validation anyway (unless the user intentionally wants to use invalid values). So I think we should validate an experiment and provide feedback after it is created or updated (in the detail page), and not during the creation or editing process (in the stepper UI).

**Comment by mswartzCL** [comment]
@zackcl Based on our conversation today about validation in the stepper and edit modals vs. displayed on the experiment detail, I am going to refine and combine stories #402 and #403.

**Comment by mswartzCL** [comment]
@zackcl What do you think about this user story now?

**Comment by zackcl** [comment]
@mswartzCL Thanks for updating the story. I think that looks good although I think we might want to still let the user know if the Context Metadata for an AppContext does not exist while the validation checkbox is enabled. Maybe we can show an error icon next to the experiment name (just like when validation failed) to indicate that something is wrong. Otherwise, the user might think that the Context Metadata exists and has been validated.

<img width="750" alt="Experiment Detail - Overview (mismatch Inactive) 10" src="https://user-images.githubusercontent.com/90279765/181598525-ac63e597-19f8-479b-be5a-d5b4d7a7a128.png">
